src/app/content/components/ContentLink.tsx

```diff
@@ -95,14 +96,30 @@ export const ContentLink = (props: React.PropsWithChildren<Props>) => {
         navigate(navigationMatch, options);
       }
     }}
+    onKeyDown={(e) => onKeyDown?.(e, async() => {
+      /*
+        All this logic has to be inside onKeyDown === Enter || Space
+      */
+      e.preventDefault();
+
+      if (hasUnsavedHighlight && !await showConfirmation(services)) {
```

Comment:
this is duplicated from the onclick, i would extract a handler used in both callbacks